File: Baekjoon/10845.js
const fs = require("fs");
const filePath = process.platform === "linux" ? "/dev/stdin" : "./input.txt";
const [n, ...words] = fs.readFileSync(filePath).toString().trim().split("\n");
let result = [];
class Queue {
  constructor() {
    this.queue = [];
    this.front = 0;
    this.rear = 0;
  }

  enqueue(value) {
    this.queue[this.rear++] = value;
  }

  dequeue() {
    if (this.front === this.rear) return undefined;
    let value = this.queue[this.front];
    delete this.queue[this.front];
    this.front++;
    return value;
  }

  peek() {
    return this.queue[this.front];
  }

  isEmpty() {
    return this.front === this.rear;
  }
}

let queue = new Queue();

words.forEach((ele) => {
  switch (ele) {
    case "pop":
      result.push(queue.isEmpty() ? -1 : queue.dequeue());
      break;

    case "size":
      result.push(queue.rear - queue.front);
      break;

    case "empty":
      result.push(queue.isEmpty() ? 1 : 0);
      break;

    case "front":
      result.push(queue.isEmpty() ? -1 : queue.peek());
      break;

    case "back":
      result.push(queue.isEmpty() ? -1 : queue.queue[queue.rear - 1]);
      break;

    default:
      let [push, num] = ele.split(" ");
      queue.enqueue(num);
      break;
  }
});

console.log(result.join("\n"));

File: python/Baekjoon/10845.py
import sys
from collections import deque

n = int(sys.stdin.readline().strip())
queue = deque()
for _ in range(n):
    line = sys.stdin.readline().strip().split()
    queueSize = len(queue)

    if line[0] == "push": queue.append(line[1])
    elif line[0] == "size": print(queueSize)
    elif line[0] == "empty": print(1 if queueSize == 0 else 0)
    elif queueSize == 0: print(-1)
    elif line[0] == "pop": print(queue.popleft())
    elif line[0] == "front": print(queue[0])
    else: print(queue[-1])